### PR TITLE
Upgrade PyArrow from 1.0.1 to 2.0.0 in CI

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes upgrade PyArrow from 1.0.1 to 2.0.0 in CI since PyArrow 2.0.0 is released now.

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ jobs:
           - python-version: 3.8
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml:117]: @itholic can we change ` pyarrow-version: 0.15.1` above to ` pyarrow-version: 1.0.1`? I think we should also test PyArrow 1.x as well.

**Comment by HyukjinKwon** [comment]
Looks good otherwise.

File: .github/workflows/master.yml (hunk 1 above)
Comment (itholic) [reply .github/workflows/master.yml:117]: Sounds good! Thanks :)

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Actually @itholic just for completeness, can you set `PYARROW_IGNORE_TIMEZONE` automatically? To keep the behaviour same looks like we should set this env for Arrow 2.0.0, see also https://github.com/apache/spark/commit/47a6568265525002021c1e5cfa4330f5b1a91469#.

I did similar stuff before at https://github.com/databricks/koalas/commit/05b93247c6384cbafbb818cd3e0b3682966a5fd8. You could refer this PR.

I was thinking we can just ignore it since we haven't officially supported nested timestamps with localization but .. after rethinking, I think it's just best to match with the Apache Spark upstream. cc @ueshin too.

**Comment by itholic** [comment]
Thanks, @HyukjinKwon . Sorry for the late.
Just make it to set the `PYARROW_IGNORE_TIMEZONE` automatically to 1 in `pyarrow` >=2.0.0.

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -125,6 +125,8 @@ jobs:
       GITHUB_OAUTH_KEY: ${{ secrets.GITHUB_TOKEN }}
       # Github Actions' default miniconda
       CONDA_PREFIX: /usr/share/miniconda
+      # Preserve legacy nested timezone behavior for pyarrow>=2, remove after SPARK-32285
+      PYARROW_IGNORE_TIMEZONE: "1",
```
Comment (ueshin) [inline .github/workflows/master.yml:129]: Do we need this?

**Review by ueshin — COMMENTED** [review]
@itholic Could you push another commit?
Seems like GitHub Action didn't run tests.

File: .github/workflows/master.yml (hunk 2 above)
Comment (itholic) [reply .github/workflows/master.yml]: Removed it, thanks for the catch!

File: databricks/koalas/__init__.py [hunk 3]
```diff
@@ -70,6 +70,26 @@ def assert_pyspark_version():
...
+        "Koalas will set it for you but it does not work if there is a Spark context already "
+        "launched."
+    )
+    os.environ["PYARROW_IGNORE_TIMEZONE"] = "1"
+elif "PYARROW_IGNORE_TIMEZONE" in os.environ:
+    raise RuntimeError(
+        "Please explicitly unset 'PYARROW_IGNORE_TIMEZONE' environment variable in both "
+        "driver and executor sides. It is required to set this environment variable only "
+        "when you use pyarrow>=2.0.0."
+    )
```
Comment (ueshin) [inline databricks/koalas/__init__.py:92]: Do we need to raise the error?

**Review by ueshin — APPROVED** [review]
Otherwise, LGTM. cc @HyukjinKwon

File: databricks/koalas/__init__.py (hunk 3 above)
Comment (itholic) [reply databricks/koalas/__init__.py:92]: Hm... I don't think it's necessary. Will remove it

**Review by HyukjinKwon — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**